snapcraft/internal/elf.py

```diff
@@ -369,7 +363,7 @@ def load_dependencies(
         # Return a set useful only for fetching libraries from the host
         library_paths = set()  # type: Set[str]
         for l in libs:
-            if os.path.exists(l.path) and not l.in_base_snap and not l.system_lib:
+            if os.path.exists(l.path) and not l.in_base_snap:
```

Comment:
This seems problematic. We're now injecting the base, which is nice, but we never actually check to see if a library is contained within it now. `l.is_base_snap` only checks the linker path (before patchelf), which is not the same as checking to see if e.g. `/usr/lib/libfoo.so` is contained within the base. As a result, every system dependency will be tracked, regardless of whether or not it's in the base.

We should probably keep this attribute, setting its value using something similar to `_get_system_libs()` (which is removed below) to check the base snap instead of the pre-generated library list.

Reply:
we have logic that still executes this check